### DISCUSSION TITLE
Onvif Tilt Conversion from Degrees to Generic Position Space

### DISCUSSION THREAD

**Discussion by mdaconta** [AxisCommunications/discussions#131] [General]
Hi,
We are using an Onvif interface to the AXIS Q6215-LE PTZ Network Camera and are having problems converting the Tilt from degrees to the Onvif Generic position space (-1.0 to 1.0).
Querying the camera via the Vapix API tells us the following:
root.PTZ.Limit.L1.MaxTilt=90
root.PTZ.Limit.L1.MinTilt=-90
So, the conversion should be straightforward. For example, an absolute Tilt of 45 degrees would be an onvif value of 0.5.
Unfortunately, we are seeing that when the camera move is executed (via onvif absoluteMove), it is exactly 1/2 of what it should be.  Of course, we can get it to work correctly by doubling the value but that is not the right solution.  
Can you tell me what I am misunderstanding?

Thanks for any assistance,
-Mike

**Comment by helalv** [comment]
Hi Mike,
What firmware version are you using in the camera? Could you also check the PTZ status by sending a GetStatus request before and after the move operation to verify that the onvif tilt values are changed as expected? 
[GetStatus.txt](https://github.com/AxisCommunications/discussions/files/10164223/GetStatus.txt)


Could you also check in the Web GUI of the camera what coordinates the camera thinks it is at by adding a text overlay with the #y modifier? e.g. When you send onvif value 0.5 does it say +45 in the overlay?

BR
Helene

**Reply by mdaconta** [reply]
Hi Helene,

The firmware version is 9.80.3.14 Build 61.

I will perform those other checks you mentioned above and get back to you...

Thanks,

-Mike

**Reply by xtrdustin** [reply]
Hi Helene,

We've found that the y coordinate in the GetStatus response matches with the Elevation value on the camera's overlay. However, the y coordinate in the PTZStatus messages in the metadata feed is half of the one in the GetStatus response. We'd prefer to use metadata streaming instead of polling GetStatus, as it gives us smoother updates of the orientation of the camera. We can temporarily work around the issue by doubling the y value when it's received, but do you know if this is fixed in a newer firmware version?

Thanks,
-Dustin

**Reply by xtrdustin** [reply]
Update: The elevation behavior seems to be the same in version 11.1.65, after we upgraded this morning. (The upgrade also deleted one of our media profiles we had set up for metadata streaming, so we had to re-create that.)

**Reply by helalv** [reply]
Hi Dustin,
Yes, I see the same behavior here. I have raised an internal ticket for further investigation regarding this. If you have a MyAxis account, we can create a ticket on our helpdesk where you can more easily follow the progress?

BR
Helene

**Reply by xtrdustin** [reply]
I have a helpdesk account, with my email dustin.king@xatorcorp.com. I don't know if it's a MyAxis account.

**Reply by helalv** [reply]
Hi again, 
A helpdesk ticket has now been created so you should be able to follow the progress if you login to [Axis helpdesk](https://www.axis.com/support/helpdesk) using the provided email.

**Reply by xtrdustin** [reply]
I see the ticket now. Thanks for looking into it

**Reply by xtrdustin** [reply]
Received a reply from Axis (Helpdesk tiket: 02042280 - Onvif Tilt Conversion from Degrees to Generic Position Space): 

> _"We have conducted a thorough re-investigation of the matter; however, we have yet to identify a viable solution that does not introduce additional complications. After careful consideration, we have made the decision not to pursue a further fix at this time."_
